### PR TITLE
fix(profile-update): form validations fixed.

### PR DESCRIPTION
**Pull request by rohitkrai03**
Fixes https://github.com/openshiftio/openshift.io/issues/3708
Fixes https://github.com/openshiftio/openshift.io/issues/3701
Fixes https://github.com/fabric8-services/fabric8-auth/issues/508
Fixes https://github.com/fabric8-services/fabric8-auth/issues/509

Fixes a few validations in profile update page.
- Profile Bio cannot be removed or set to empty string.
- Company name cannot be removed.
- URL cannot be removed.
- URL validation is not working correctly. For example i can update URL to http:// and it gets updated.
- Also, removed imageUrlValidation since we don't take user input for that in new design.

### CODE REVIEW

File: src/app/profile/update/update.component.spec.ts [hunk 1]
```diff
@@ -358,26 +358,9 @@ describe('UpdateComponent', () => {
...
-      component.imageUrl = url;
-      component.validateImageUrl();
-      expect(component.imageUrlInvalid).toBeFalsy();
-    });
-  });
-
   describe('#validateUrl', () => {
     it('should verify not-a-real-url to be an invalid url', () => {
-      let url: string = 'not-a-real-url';
+      let url: string = 'htt:/not-a-real-url.some-thing.';
```
Comment (jiekang) [inline src/app/profile/update/update.component.spec.ts]: Could you add an array of strings containing various invalid urls and then run the inner 'test' over all of them?

**Comment by jiekang** [comment]
For the point 'URL validation is not working correctly. For example i can update URL to http:// and it gets updated.', I'm a little confused. In the regex tester it looks like this would not match http:// so it should return the correct value. Are you saying that even though this validation function works, it still lets you update the URL? Or this validation function doesn't work for this case?

**Comment by rohitkrai03** [comment]
@jiekang It doesn't let you update the URL to `http://`. The description just gives a summary of all the validation problems this PR fixes. That may be the reason for the confusion. I've also added this use case in the test.

**Comment by fabric8cd** [comment]
@rohitkrai03 fabric8/fabric8-ui:SNAPSHOT-PR-2978-3 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2978-fabric8-ui.badger.fabric8.io

**Pull request merged**